### PR TITLE
Rewrite uri module to improve ergonomics.

### PR DESCRIPTION
**Pull request by msmorgan**
- Make clippy happy.
- Use url crate for constructing urls.
- Update list of formats.
- Rewrite `uri` module, add `list` object type.
- Bump version number.
- Add list::PageIter.
- Move uri module to root and make Uri public.

## Undecided:

It might be valid to change `<ListIter<T> as Iterator>::Item` to `T` instead of `crate::Result<T>`.
The first `ListIter` will be wrapped in a `Result`, but the subsequent ones should be created from known-good
URLs and therefore shouldn't error. Using `unwrap()` seems like it could be justified, but it might be
better for users if we print a warning and simply return `None` in these cases. The same could be said about
`PageIter`. What do you think?

### CODE REVIEW

**Comment by msmorgan** [comment]
The last commit ([2cdd5d3](https://github.com/mendess/scryfall-rs/pull/16/commits/2cdd5d3ea5f1fe291e9ebbbbdd5844633027dfc1)) has the change from `Item = crate::Result<T>` to `Item = T`. Let me know what you think. I think the ease of use is worth making next-page response errors un-handle-able. Plus, there are other ways to robustly get each page and handle all errors. See `Uri::fetch_all` for an example of this.

File: src/list.rs [hunk 1]
```diff
@@ -0,0 +1,202 @@
...
+    }
+}
+
+impl<T: DeserializeOwned> IntoIterator for List<T> {
+    type IntoIter = ListIter<T>;
+    type Item = T;
+
+    fn into_iter(self) -> Self::IntoIter {
+        // `has_more` is assumed to be redundant.
+        debug_assert!(self.has_more == self.next_page.is_some());
```
Comment (mendess) [inline src/list.rs:46]: maybe you should have `has_more` as just a method that checks the option and remove it from the struct?

**Review by mendess — APPROVED** [review]

**Comment by mendess** [comment]
Really good work you did here. I remember wanting to do the construction of the urls without using `format!` but I never got around to it

File: src/list.rs (hunk 1 above)
Comment (msmorgan) [reply src/list.rs:46]: I included it so the struct would have all the fields from the [List Object](https://scryfall.com/docs/api/lists) page in the docs.

**Pull request merged**